pymeasure/adapters/ni_gpib_232ct.py

```diff
@@ -0,0 +1,346 @@
+
+    @eoi.setter
+    def eoi(self, value):
+        self.connection.write(f"EOT {int(value)}")
+
+    def gpib(self, address, **kwargs):
+        """Return a NI_GPIB_232 object that references the GPIB
+        address specified, while sharing the Serial connection with other
+        calls of this function.
+
+        :param address: Integer GPIB address of the desired instrument
+        :param kwargs: Arguments for the initialization
+        :returns: NI_GPIB_232 for specific GPIB address
+        """
+        return NI_GPIB_232(self, address, **kwargs)
+
+    def pass_control(self, primary_address: int, secondary_address: int):
+        """
+        Pass control to device with primary_address and optional secondary_address.
+
+        """
+        self.connection.write(f"pct {primary_address}+{secondary_address}")
+
+    def send_command(self, data: bytes):
+        """
+        Write GPIB command bytes on the bus.
+
+        """
+        self.connection.write(f"cmd #{len(data)}\n {data}")
+        self._check_errors()
+
+    def set_rsc(self):
+        """
+        Set the NI-GPIB232ct to become the GPIB system controller.
+
+        """
+        self.connection.write("rsc 1")
+
+    def send_ifc(self):
+        """Pulse the interface clear line (IFC) for at least 200 microseconds."""
+        self.connection.write("sic 0.0002")
+
+    @property
+    def timeout(self):
+        """Control the GPIB timeout.
+        Valid value range 0.0001 to 3600s
+        """
+        self.connection.write("tmo")
+        ret_val = self.connection.read().split(",")
+        return float(ret_val[0])
+
+    @timeout.setter
+    def timeout(self, value):
+        value = strict_range(value, [0.0001, 3600])
+        if value < 1:
+            self.connection.write(f"tmo {value:.4f}")
+        elif value <= 3600:
+            self.connection.write(f"tmo {int(value)}")
+
+    @property
+    def version(self):
+        """Get the version string of the NI GPIB-232-CT."""
+        self.flush_read_buffer()
+        self.connection.write("id \r")
+        time.sleep(0.02)
+        return self.connection.read_bytes(71).decode()
+
+    def wait_for_srq(self, timeout=20, delay=0.1):
+        """Blocks until a SRQ, and leaves the bit high
+
+        :param timeout: Timeout duration in seconds.
+        :raises TimeoutError: "Waiting for SRQ timed out."
+        """
+        stop = time.perf_counter() + timeout
+        while time.perf_counter() < stop:
+            self.connection.write(f"rsp {self.address}")
+            if self.connection.bytes_in_buffer == 0:
+                time.sleep(delay)
+            if self.connection.bytes_in_buffer >= 1:
+                ret_val = self.connection.read_bytes(self.connection.bytes_in_buffer)
+                if int(ret_val) > 0:
+                    return int(ret_val)
+                if int(ret_val) == -1:
+                    raise TimeoutError(f"Waiting for SRQ timed out after {timeout}")
```

Comment:
Is this a different timeout than the one controlled by the `timeout` argument for this method? If yes, this could probably use some clarification.
I would have expected the TimeoutError to be raised when the while loop exited because the wall time went above the `timeout` cutoff (i.e. that is currently `return None`).

Reply:
Yes, it is a dedicate timeout paramter for the SRQ "scan"

Reply:
I think it's a bit confusing that there are different timeout conditions. Does the difference matter between a timeout from the loop and a timeout from the scan?